app/views/foreman/job_templates/change_content_source.erb

```diff
@@ -0,0 +1,46 @@
+<%#
+kind: job_template
+name: Change content source
+job_category: Katello
+model: JobTemplate
+provider_type: SSH
+description_format: Configure subscription manager to new content source (smart proxy)
+feature: katello_change_content_source
+template_inputs:
+- name: smart_proxy
+  required: true
+  input_type: user
+  value_type: resource
+  resource_type: SmartProxy
+  description: New content source (Smart Proxy)
+  advanced: false
```

Comment:
IMHO this template should not have any inputs. It should use `host.content_source` as the Smart Proxy.

Reply:
That would not work in case of rendering `app/views/foreman/job_templates/change_content_source_and_host.erb` template.

When `change_content_source_and_host.erb` template is rendered, `host.content_source` is still the old one, while `smart_proxy` input is the new one.

So running `change_content_source_and_host` job would not configure `subscription-manager` to the new proxy but to the old one.

Reply:
> That would not work in case of rendering `app/views/foreman/job_templates/change_content_source_and_host.erb` template.

Then I'd recommend to extracted it to a snippet that is properly reusable.

Reply:
> Then I'd recommend to extracted it to a snippet that is properly reusable.

But we want to have it as job, so users can make changes on the hosts after they changed content source in UI. We can't run snippets as job right?

Reply:
I was suggesting to extract the actual logic to a snippet and then create two job templates.